---
title: "Data sources"
output: html_document
---

```{r loadLibraries, message=FALSE, include=FALSE}
library(readxl)
```

```{r defFun, echo=F}
myDownload <- function(URL,
                       dstPath = "indata",
                       dstFile = file.path(dstPath, basename(URL))){
  if( !file.exists(dstFile) )
    download.file(URL, dstFile,quiet = T)
  cat(sprintf("* [%s](%s)\n",basename(dstFile),URL))
  return(dstFile)
}

```

### From PODC

Expression matrices:

```{r PODCexpMat, echo=F, results='asis'}

PODCexpMatURLs <- 
  c( # "http://plantomics.mind.meiji.ac.jp/podc/data/zea_mays_fpkm.tsv.gz",
    "http://plantomics.mind.meiji.ac.jp/podc/data/arabidopsis_thaliana_fpkm.tsv.gz",
    "http://plantomics.mind.meiji.ac.jp/podc/data/oryza_sativa_fpkm.tsv.gz",
    "http://plantomics.mind.meiji.ac.jp/podc/data/solanum_lycopersicum_fpkm.tsv.gz",
    "http://plantomics.mind.meiji.ac.jp/podc/data/sorghum_bicolor_fpkm.tsv.gz",
    "http://plantomics.mind.meiji.ac.jp/podc/data/vitis_vinifera_fpkm.tsv.gz",
    "http://plantomics.mind.meiji.ac.jp/podc/data/medicago_truncatula_fpkm.tsv.gz",
    "http://plantomics.mind.meiji.ac.jp/podc/data/solanum_tuberosum_fpkm.tsv.gz",
    "http://plantomics.mind.meiji.ac.jp/podc/data/glycine_max_fpkm.tsv.gz",
    "http://plantomics.mind.meiji.ac.jp/podc/data/nicotiana_tabacum_fpkm.tsv.gz"
    )

invisible( sapply(PODCexpMatURLs, myDownload) )

```

Sample metadata:

```{r PODCmetadata, echo=F, results='asis'}

PODCmetadataURLs <- 
  c("http://plantomics.mind.meiji.ac.jp/podc/data/arabidopsis_thaliana_sample.xls",
    "http://plantomics.mind.meiji.ac.jp/podc/data/oryza_sativa_sample.xls",
    "http://plantomics.mind.meiji.ac.jp/podc/data/solanum_lycopersicum_sample.xls",
    "http://plantomics.mind.meiji.ac.jp/podc/data/sorghum_bicolor_sample.xls",
    "http://plantomics.mind.meiji.ac.jp/podc/data/vitis_vinifera_sample.xls",
    "http://plantomics.mind.meiji.ac.jp/podc/data/medicago_truncatula_sample.xls",
    "http://plantomics.mind.meiji.ac.jp/podc/data/solanum_tuberosum_sample.xls",
    "http://plantomics.mind.meiji.ac.jp/podc/data/glycine_max_sample.xls",
    "http://plantomics.mind.meiji.ac.jp/podc/data/nicotiana_tabacum_sample.xls",
    "http://plantomics.mind.meiji.ac.jp/podc/data/zea_mays_sample.xls"    
    )

PODCmetadataFiles <- sapply(PODCmetadataURLs, myDownload)

```

Note: The "readxl" package couldn't open the .xls files, but managed to open .xlsx files. The .xls files therefore had to be converted .xlsx, which was done "manually". For convenience xls files were converted to .RDS files

### Download RNAseq data from EBI

Since the Zm and Gm data used in PODC was a different genome annotation than the one at Ensembl, expression data for the same samples is downloaded from EBI using the API described at http://www.ebi.ac.uk/fg/rnaseq/api/ 


```{r downloadRNAseqFromEBI, echo=FALSE, results='asis'}
source("R/ebiRNAseqAPI.R")



downloadRNAseq <- function(organism,dstPath,PODCmetadataFile = paste0("indata/",organism,"_sample.xlsx")){
  studyDataFile <- file.path(dstPath,"studyData.tsv")
  #
  # get study data which include the url for the expression data
  #
  if(!file.exists(studyDataFile)){
    # read the PODC sample data
    sampleData <- read_excel(PODCmetadataFile,col_types = rep("text",15))
    
    # get all studies with RNAseq data for Zea mays
    studyData <- RNAseq_getStudiesByOrganism( organism )
    
    studyData <- studyData[studyData$STUDY_ID %in% sampleData$Study, ]
  
    studyData <- studyData[studyData$STATUS == "Complete", ]
    
    cat( "Found expression data for",sum(sampleData$Study %in% studyData$STUDY_ID),
         "of",nrow(sampleData),"samples\n")
  
    dir.create(dstPath)
    write_tsv(studyData,path = studyDataFile)
  } else {
    studyData <- read_tsv(studyDataFile,col_types = cols())
  }
  
  
  # the fpkm url is not among the data but it has the same path as the CRAM url
  FPKMurls <- studyData$GENES_FPKM_COUNTS_FTP_LOCATION
  dstFiles <- file.path(dstPath,paste(studyData$STUDY_ID,basename(FPKMurls),sep = "."))
  
  #
  # Download
  #
  invisible(mapply(
    FUN = myDownload, 
    URL = FPKMurls, 
    dstFile = dstFiles))
  
}

invisible(mapply(
  FUN = downloadRNAseq,
  organism=c("glycine_max","zea_mays"),
  dstPath=c("indata/RNAseqGm","indata/RNAseqZm")
))


```


### Ensembl BioMart

* At_Os_orthologs.txt - At:Os orthologs
* Os_At_orthologs.txt - Os:At orthologs

```{r biomart, eval=F, include=F}
library(biomaRt)

At_Os_ortho_file <- "../indata/At_Os_orthologs.txt"

# download from biomart if not already downloaded
if( !file.exists(At_Os_ortho_file) ){
  library(biomaRt)
  
  # listMarts(host="plants.ensembl.org")
  
  # use arabidopsis database
  ensembl <- useMart("plants_mart",dataset="athaliana_eg_gene", host="plants.ensembl.org")
  
  # get At gene id and corresponding Os ortholog. Filter: only genes with Os homolog
  At_Os_orthos <- getBM(attributes=c("ensembl_gene_id", 
                                     "osativa_eg_homolog_ensembl_gene",
                                     "osativa_eg_homolog_orthology_type",
                                     "osativa_eg_homolog_subtype",
                                     "osativa_eg_homolog_perc_id",
                                     "osativa_eg_homolog_perc_id_r1"), 
                        filters = 'with_osativa_eg_homolog', values = T, mart = ensembl)

  # change the column names
  colnames(At_Os_orthos) <- c("At","Os","otype","MRCA","PID","PIDr")  
  # convert orthology type
  oTypes <- c(ortholog_many2many = "N:N", ortholog_one2many = "1:N", ortholog_one2one = "1:1")
  At_Os_orthos$otype <- oTypes[At_Os_orthos$otype]

  write.table(At_Os_orthos,file = At_Os_ortho_file, row.names = F,quote = F,sep="\t")
}

#
# Get Os:At orthologs also
#
Os_At_ortho_file <- "../indata/Os_At_orthologs.txt"

# download from biomart if not already downloaded
if( !file.exists(Os_At_ortho_file) ){
  
  ensembl <- useMart("plants_mart",dataset="osativa_eg_gene", host="plants.ensembl.org")
  
  # get Os gene id and corresponding At ortholog. Filter: only genes with At homolog
  Os_At_orthos <- getBM(attributes=c("ensembl_gene_id", "athaliana_eg_homolog_ensembl_gene","athaliana_eg_homolog_subtype"), 
                        filters = 'with_athaliana_eg_homolog', values = T, mart = ensembl)
  
  colnames(Os_At_orthos) <- c("Os","At","MRCA")  
  
  write.table(Os_At_orthos,file = Os_At_ortho_file, row.names = F,quote = F,sep="\t")
}

```

```{r}
library(biomaRt)

downloadParalogs <- function(species="athaliana", filename){
  # use the ensembl plants mart
  ensembl <- useMart("plants_mart",dataset=paste0(species,"_eg_gene"), host="plants.ensembl.org")
  # get gene id, paralogs and MRCA
  paralogs <- getBM(attributes=c("ensembl_gene_id", 
                                 paste0(species,"_eg_paralog_ensembl_gene"),
                                 paste0(species,"_eg_paralog_subtype")), # MRCA
                    mart = ensembl)
  # change the column names
  colnames(paralogs) <- c("geneID","paralog","MRCA")
  
  saveRDS(paralogs,file = filename)
}

At_paralog_file <- "../indata/At_paralogs.RDS"
Os_paralog_file <- "../indata/Os_paralogs.RDS"

# download from biomart if not already downloaded
if( !file.exists(At_paralog_file) ){
  downloadParalogs(species = "athaliana", filename = At_paralog_file)
}
if( !file.exists(Os_paralog_file) ){
  downloadParalogs(species = "osativa", filename = Os_paralog_file)
}


# download from biomart if not already downloaded

```

### Ensembl compara gene trees

Gene tree dump from Ensembl Plants version 33.

```{r ComparaGeneTrees, echo=F, results='asis'}
myDownload(URL = "ftp://ftp.ensemblgenomes.org/pub/plants/release-33/emf/ensembl-compara/homologies/Compara.phyloxml_aa_trees.33.tar.gz")
```
