<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>CoExCorr</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-1.1/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CoExCorr project</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="DuplicateGenes.html">Duplicate genes</a>
</li>
<li>
  <a href="WithinSpecies.html">Within-species ranks</a>
</li>
<li>
  <a href="CompareMethods.html">Alternative methods</a>
</li>
<li>
  <a href="downloads.html">Data sources</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">CoExCorr</h1>

</div>


<div id="co-expression-correlation---a-method-for-cross-species-comparative-transcriptomics." class="section level2">
<h2>Co-expression correlation - A method for cross-species comparative transcriptomics.</h2>
<p>This study explores a method for indirectly comparing gene expression data from different species by correlating the co-expression of orthologous genes.</p>
<p>The final output of the algorithm is a measure of expression similarity between any two genes in two different species. These are the steps:</p>
<ul>
<li>MI: Calculate mutual information (MI) co-expression matrix for each species.</li>
<li>CLR: “Normalize” the co-expression using the CLR algorithm.</li>
<li>Coexpression Correlation Score (CCS): Compare co-expression matrices by calculating the pearson correlation coefficient (PCC) between co-expression matrices using a set of 1:1 orthologs as reference.</li>
</ul>
<div id="data-sources" class="section level3">
<h3>Data sources</h3>
<p>To test the algorithm we used public gene expression data from five different plants:</p>
<ul>
<li>Arabidopsis thaliana (At)</li>
<li>Solanum lycopersicum (Sl)</li>
<li>Glycine max (Gm)</li>
<li>Oryza sativa (Os)</li>
<li>Zea mays (Zm)</li>
</ul>
<p>The species were chosen because of the amount of available expression data. All the data used has been downloaded from public databases, details on the <a href="downloads.html">data sources</a> page.</p>
</div>
<div id="ccs-performance-evaluation" class="section level3">
<h3>CCS Performance evaluation</h3>
<p>To evaluate the perfomance of the algorithm we calculate, for each ortholog pair, the rank of the CCS of the ortholog compared to the CCS of all non-orthologs of the gene. This rank is normalised to get a value between 0 and 1 (which we call the ortholog ranks score (ORS)) where 0 represents the highest CCS and 1 the lowest. The idea is that the orthologs should have a more similar expression pattern, i.e. higher CCS, than when comparing with a random gene. ORS can also be interpreted as one minus the P-value (i.e. ORS = 1-P), and is plotted as -log10(P).</p>
<p>As the different species are expected to have differences in gene regulation, we also test the score within the same species by dividing the expression data in two subsets and compare them as if they were from different species. This way we test only the effect of the method and the effect of the sample choice. See <a href="WithinSpecies.html">within-species ranks</a>.</p>
</div>
<div id="gene-regulation-divergence-of-duplicated-genes" class="section level3">
<h3>Gene regulation divergence of duplicated genes</h3>
<p>In the <a href="DuplicateGenes.html">duplicate genes</a> analysis we look at the expression similarity of 1-to-many and many-to-many orthologs, i.e. orthologs with duplications.</p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
